### PR TITLE
Feature: update events sf43

### PR DESCRIPTION
**Pull request by ayshiff**
This PR updates the way events are used since Symfony 4.3 Beta 2 [#975](https://github.com/symfony/demo/issues/975)

- removed Event class
- created a `CommentCreatedEvent` event object
- update calls to dispatch()
- used `Symfony\Contracts\EventDispatcher\Event` instead of the `Event` class

I have a problem when I update the calls to dispatch() to remove the event name. The `comment.created` event is inside the `not called listeners` but it works fine when I keep the event name.   
I don't know if it is an issue or not.

### CODE REVIEW

File: src/Controller/BlogController.php [hunk 1]
```diff
@@ -109,14 +108,17 @@ public function commentNew(Request $request, Post $post, EventDispatcherInterfac
...
+            // https://symfony.com/blog/new-in-symfony-4-3-simpler-event-dispatching
+            $event = new CommentCreatedEvent($comment);
 
             // When an event is dispatched, Symfony notifies it to all the listeners
             // and subscribers registered to it. Listeners can modify the information
             // passed in the event and they can even modify the execution flow, so
             // there's no guarantee that the rest of this controller will be executed.
             // See https://symfony.com/doc/current/components/event_dispatcher.html
-            $eventDispatcher->dispatch(Events::COMMENT_CREATED, $event);
+            $eventDispatcher->dispatch($event, CommentCreatedEvent::NAME);
```
Comment (yceruto) [inline src/Controller/BlogController.php]: The best part of this new feature is that we can omit the second argument here

File: src/EventSubscriber/CommentNotificationSubscriber.php [hunk 2]
```diff
@@ -41,14 +40,14 @@ public function __construct(\Swift_Mailer $mailer, UrlGeneratorInterface $urlGen
     public static function getSubscribedEvents(): array
     {
         return [
-            Events::COMMENT_CREATED => 'onCommentCreated',
+            CommentCreatedEvent::NAME => 'onCommentCreated',
```
Comment (yceruto) [inline src/EventSubscriber/CommentNotificationSubscriber.php]: using `CommentCreatedEvent::class` instead

File: src/Events/CommentCreatedEvent.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+use App\Entity\Comment;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * The comment.created event is dispatched each time a comment is created
+ * in the system.
+ */
+class CommentCreatedEvent extends Event
+{
+    public const NAME = 'comment.created';
```
Comment (yceruto) [inline src/Events/CommentCreatedEvent.php]: Now it's useless

**Comment by yceruto** [comment]
The composer.json needs to be updated to requiere 4.3-beta2 and make tests pass

**Comment by ayshiff** [comment]
Maybe I should wait that [#974](https://github.com/symfony/demo/pull/974) is merged.

**Comment by javiereguiluz** [comment]
@ayshiff thanks for this contribution! We did some minor tweaks while merging, but your pull request was great. Thanks to reviewers too.

**Pull request closed without merging**